validate.go

```diff
@@ -192,6 +194,48 @@ func checkProcess(process rspec.Process, rootfs string) {
 	}
 }
 
+func supportedMountTypes() (map[string]bool, error) {
+	supportedTypes := make(map[string]bool)
+	f, err := os.Open("/proc/filesystems")
```

Comment:
On Thu, Jun 23, 2016 at 07:55:48PM -0700, Ma Shimiao wrote:
> But currently supportedMountTypes() just supports Linux and I'm not
> sure does other systems has file like /proc/filesystems to list
> system supported file systems?

The problem with /proc/filesystems is partly that only a Linux host
will have it, but also that the contents depend on the particular
kernel.  For example, I have ‘nfs4’ in my /proc/filesystems because my
kernel was compiled with CONFIG_NFS_V4=y.  My kernel was not compiled
with CONFIG_SQUASHFS, so I don't have ‘squashfs’ in my
/proc/filesystems.  So this, like whether or not a hook command is
present [1], is a check that may pass on some hosts but not others.
It's important if you're about to run that config on that host, but
less useful if you're running a validation service or some such.

So I think we want:

* A --host-specific flag that skips this test (at least on Linux,
  where the test is based on a per-kernel /proc/filesystems).
* A platform.os check, because we don't want to be comparing mounts
  against /proc/filesystems if the container targets Windows.
* A skip if platform.os is ‘linux’, but we can't open
  /proc/filesystems (e.g. because we're running the validator on a
  Windows host).

[1]: https://github.com/opencontainers/ocitools/pull/49#issuecomment-216585596